

## Gao et al. 2015 (transcriptomics of Haematococcus pluvialis)

- used gene2go to get GO terms

	ftp://ftp/ncbi.nih.gov/DATA/gene2go.gz

- also mapped genes to KEGG pathways

- generated stats on gene categories (how many genes are in signal transduction pathways, how many metabolic genes, etc...)
- generated stats on the number of genes identified

- used GoPipe to analyze the expression data

-  transcription factors (TFs) were identified by searching the Arabidopsis Gene Regula- tory Information Server (AGRIS) Database [12].


## Rowe et al. 

- used tophat2 to align reads
- used DESeq to count reads (count based strategy)
- annotation using the KEGG KAAS server


## Osborn et al.

- used custom microarrays


## Simon et al.

- SOLiD sequencing
- used MAQ to align reads
- DESeq and read counting
- because this is chlamy, they simply retreived GO and KEGG from the databases directly


## Kim et al.

- used 454 sequencing
- seems like manual assembly and annotation


## Nabity et al.

- no bioinformatics done


## Gao et al. 2012

- RT PCR


## Atkinson et al.

- nice review paper listing some genes that might be of interest to Desmoddesmus


## Urano et al.

- another review paper discussing genes and pathways involved. 
- has diagram discussing transcriptional events


## Jamers et al.

- another review paper





